src/classes/HouseholdNamingService.cls

```diff
@@ -0,0 +1,512 @@
+        for (SObject household : records) {
+            if (needsNamingFieldsUpdated(household, oldMap.get(household.Id))) {
+                hhList.add(household.Id);
+            }
+        }
+        return hhList;
+    }
+
+    public void setHouseholdNameFields(List<SObject> records, Map<Id, SObject> oldMap) {
+        List<Id> ids = new List<Id>();
+        for (SObject sobj : records) {
+            ids.add(sobj.Id);
+        }
+        Map<Id, List<Contact>> contactsByHouseholdId =
+                getHouseholdMembersByHouseholdId(ids);
+
+        for (SObject household : records) {
+            SObject oldRecord = oldMap.get(household.Id);
+            setCustomNamingStringValue(household, oldRecord);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of Household or Account Ids, update their Household Names
+    * @param accountIds the list of Household or Account Ids that need updating.
+    * @return void
+    */
+    public void renameHouseholdAccounts(List<Id> accountIds) {
+        renameHHAccounts(accountIds);
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of Household or Account Ids, update their Household Names
+    * @param accountIds the list of Household or Account Ids that need updating.
+    * @return void
+    */
+    public static void renameHHAccounts(List<Id> accountIds) {
+        if (accountIds.isEmpty()) {
+            return;
+        }
+
+        if (shouldProcessSynchronously(accountIds)) {
+            HouseholdNamingService householdNamingService = new HouseholdNamingService();
+            householdNamingService.updateHouseholdNameAndMemberCount(accountIds);
+        } else {
+            HouseholdNamingService.updateHouseholdNameAndMemberCountAsynchronously(accountIds);
+        }
+    }
+
+    private static Boolean shouldProcessSynchronously(List<Id> accountIds) {
+        return accountIds.size() == 1 || System.isFuture() || System.isBatch() || System.isQueueable();
+    }
+
+    public void setNameFieldValuesOnHousehold(SObject household, HouseholdName householdName) {
+        HouseholdCustomNamingString customNamingString =
+                new HouseholdCustomNamingString(customNamingStringFor(household));
+        if (!customNamingString.contains('Name')) {
+            household.put('Name', householdName.value());
+        }
+        if (!customNamingString.contains('Formal_Greeting__c')) {
+            household.put('npo02__Formal_Greeting__c', householdName.asFormalGreeting());
+        }
+        if (!customNamingString.contains('Informal_Greeting__c')) {
+            household.put('npo02__Informal_Greeting__c', householdName.asInformalGreeting());
+        }
+    }
+
+    public HouseholdCustomNamingString getCustomNamingString(SObject household, SObject oldRecord) {
```

Comment:
Since we expect to get a custom naming string here, Do we think we can return a string here instead of the instance of the class?